src/components/input/Button/Button.styles.js

```diff
@@ -1,65 +1,85 @@
 import styled, { css } from 'styled-components';
 
 import { buttonColors } from '../../../base/color/colorVariables';
-import fontSizes from '../../../base/typography/fontSizes';
+import { interfaceTypography } from '../../../base/typography/typographyVariables';
+import spacerSizes from '../../../base/spacing';
 
-export const buttonModes = {
+
+// We're going to try to colour the buttons based on state or intent
+export const setButtonColour = (intent) => {
+  console.log(intent);
+  //const currentButtonColour = buttonColors[intent];
+  console.log(currentButtonColour);
+  return currentButtonColour;
```

Comment:
This is referencing an undefined variable (`currentButtonColour`) so it won't work. You'll need to uncomment the code above for this to work.

Reply:
Yes! This is currently commented out while I'm trying to make everything work, but I'm stuck on things before this that I can't figure out how to fix—I can't test this function yet, because I can't figure out how to call it from within the structure I'm using.